src/customers/service/customers.service.ts

```diff
@@ -14,7 +19,21 @@ export class CustomersService {
     return customer;
   }
 
-  createCustomer(customer: Customer): Promise<Customer> {
-    return this.customersRepository.create(customer);
+  async getCustomersByUsername(username: string) {
+    return await this.customersRepository.findOneByUsername(username);
+  }
+
+  // async createCustomer(customer: Customer): Promise<Customer> {
+  //   return await this.customersRepository.create(customer);
+  // }
+
+  async createCustomer(customer: Customer): Promise<Customer> {
+    try {
+      const hashedPassword = await hash(customer.password, 10);
+      customer.password = hashedPassword;
```

Comment:
Can be done shorter like `customer.password = await hash(customer.password, 10);`

Reply:
done